File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
answer = 0

def solution(numbers, target):

    dfs(numbers, 0, target, 0)

    return answer


def dfs(numbers, depth, target, sum):
    if depth == len(numbers):
        if sum == target:
            global answer
            answer += 1
    else:
        dfs(numbers, depth + 1, target, sum + numbers[depth])
        dfs(numbers, depth + 1, target, sum - numbers[depth])

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    
    private static int answer = 0;
    
    public int solution(int[] numbers, int target) {
        
        dfs(numbers, 0, target, 0);
        
        return answer;
    }
    
    public void dfs(int[] numbers, int depth, int target, int sum) {
        if(depth == numbers.length) {
            if(target == sum) answer++;
        } else {
            dfs(numbers, depth + 1, target, sum + numbers[depth]);
            dfs(numbers, depth + 1, target, sum - numbers[depth]);
        }
    }
    
    
    
    
}